src/executor/validate.rs

```diff
@@ -0,0 +1,245 @@
+use im_rc::HashSet;
+use serde::Serialize;
+use sqlparser::ast::{ColumnDef, ColumnOption, DataType, Ident};
+use std::{convert::TryInto, fmt::Debug, rc::Rc};
+use thiserror::Error as ThisError;
+
+use crate::data::{Row, Schema, Value};
+use crate::result::Result;
+use crate::store::Store;
+use crate::utils::Vector;
+
+pub enum ColumnValidation {
+    All,
+    SpecifiedColumns(Vec<Ident>),
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum UniqueKey {
+    Bool(bool),
+    I64(i64),
+    Str(String),
+    Null,
+}
+
+#[derive(Debug, PartialEq, Serialize, ThisError)]
+pub enum ValidateError {
+    #[error("conflict! storage row has no column on index {0}")]
+    ConflictOnStorageColumnIndex(usize),
+
+    #[error("conflict! column '{0}' of data type '{1}' conflicts with unique constraint")]
+    ConflictOnUniqueColumnDataType(String, String),
+
+    #[error("duplicate entry '{0}' for unique column '{1}'")]
+    DuplicateEntryOnUniqueField(String, String),
+
+    #[error("table already exists")]
+    TableAlreadyExists,
+
+    #[error("table does not exist")]
+    TableNotExists,
```

Comment:
There is a test case for `ExecuteError::TableNotExists` but no `ValidateError::TableNotExists`.
Please add a test case which generates `ValidateError::TableNotExists`.